src/Crocodealer/Command/Sync.hs

```diff
@@ -0,0 +1,81 @@
+module Crocodealer.Command.Sync
+       ( sync
+       ) where
+
+import Network.HTTP.Simple (Response, getResponseBody, getResponseStatus, httpBS, parseRequest)
+import Network.HTTP.Types.Status (Status (..))
+
+import Crocodealer.Command.Colors (errorMessage, infoMessage, warningMessage)
+
+data File = File
+    { fileName     :: !Text
+    , fileContents :: !Text
+    }
+
+data FileStatus
+    = UpToDate
+    | Outdated
+    | FileDoesNotExist
+
+data Repo = Repo
+    { repoUser   :: !Text
+    , repoName   :: !Text
+    , repoStatus :: !FileStatus
+    }
```

Comment:
I don't think that `FileStatus` should be a part of the `Repo` data type. I think there should be no `Repo` data type and I imagine the following forkflow:

```haskell
-- | This function tries to fetch file content and returns `Nothing` if fails
fetchGitHubFile :: UserName -> RepoName -> FileName -> IO (Maybe ByteString)
```

Then the main function could be

```haskell
sync :: UserName -> RepoName -> FileName -> [RepoName] -> IO ()
sync ... = do
    templateFile <- fetchTemplateFile
    forM_ repos $ \repo -> do
        repoFile <- fetchGitHubFile repo fileName
        fileDiff = calculateDiff templateFile repoFile
        printFileDiff fileDiff
```

Something like this. So you process repos one by one and you print the diff immediately when you download the file. Does this make sense?

Reply:
Yeah, it will definitely remove some of the complexity from the code.